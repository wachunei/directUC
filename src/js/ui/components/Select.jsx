import styled from "styled-components";

const Select = styled.select`
  box-sizing: border-box;
  appearance: none;
  border: 1px solid ${(props) => props.theme.colors.inputBorder};
  background: white;
  padding: 0.5rem 1.6rem 0.5rem 0.5rem;
  border-radius: 0;
  background-image: url("data:image/svg+xml,%3Csvg width='13' height='13' viewBox='0 0 10 10' fill='none' xmlns='http://www.w3.org/2000/svg'%3E %3Cpath d='M9.48486 4.24268L5.24219 8.48535L0.999512 4.24268' stroke='%23909090'/%3E %3C/svg%3E ");
  background-repeat: no-repeat;
  background-position: right 0.5rem top calc(50% - 1px);
  transition: filter 0.05s;

  &:focus:not(:disabled) {
    border: 1px solid ${(props) => props.theme.colors.primary};
  }

  :disabled {
    cursor: not-allowed;
    filter: grayscale(100%) opacity(60%);
  }
`;

export default Select;
