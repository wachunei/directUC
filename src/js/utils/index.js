export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
