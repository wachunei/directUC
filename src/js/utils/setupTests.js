// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
