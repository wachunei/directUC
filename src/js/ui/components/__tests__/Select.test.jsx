import React from "react";
import Select from "../Select";
import { renderTree } from "./utils";
import "jest-styled-components";

describe("Select Component", () => {
  test("should match snapshot", () => {
    const tree = renderTree(<Select />);
    expect(tree).toMatchSnapshot();
  });

  test("should match snapshot with lightTheme", () => {
    const tree = renderTree(<Select />, "lightTheme");
    expect(tree).toMatchSnapshot();
  });

  test("should match snapshot with darkTheme", () => {
    const tree = renderTree(<Select />, "darkTheme");
    expect(tree).toMatchSnapshot();
  });

  test("should accept options", () => {
    const tree = renderTree(
      <Select>
        <option disabled hidden value="">
          option
        </option>
      </Select>
    );
    expect(tree).toMatchSnapshot();
  });
});
