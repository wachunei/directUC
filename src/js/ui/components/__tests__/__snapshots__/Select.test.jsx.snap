// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component should accept options 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #909090;
  background: #FFFFFF;
  padding: 0.5rem 1.6rem 0.5rem 0.5rem;
  border-radius: 0;
  background-image: url("data:image/svg+xml,%3Csvg width='13' height='13' viewBox='0 0 10 10' fill='none' xmlns='http://www.w3.org/2000/svg'%3E %3Cpath d='M9.48486 4.24268L5.24219 8.48535L0.999512 4.24268' stroke='%23909090'/%3E %3C/svg%3E ");
  background-repeat: no-repeat;
  background-position: right 0.5rem top calc(50% - 1px);
  -webkit-transition: filter 0.05s;
  transition: filter 0.05s;
}

.c0:focus:not(:disabled) {
  border: 1px solid #1F95EB;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(60%);
  filter: grayscale(100%) opacity(60%);
}

<select
  class="c0"
>
  <option
    disabled=""
    hidden=""
    value=""
  >
    option
  </option>
</select>
`;

exports[`Select Component should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #909090;
  background: #FFFFFF;
  padding: 0.5rem 1.6rem 0.5rem 0.5rem;
  border-radius: 0;
  background-image: url("data:image/svg+xml,%3Csvg width='13' height='13' viewBox='0 0 10 10' fill='none' xmlns='http://www.w3.org/2000/svg'%3E %3Cpath d='M9.48486 4.24268L5.24219 8.48535L0.999512 4.24268' stroke='%23909090'/%3E %3C/svg%3E ");
  background-repeat: no-repeat;
  background-position: right 0.5rem top calc(50% - 1px);
  -webkit-transition: filter 0.05s;
  transition: filter 0.05s;
}

.c0:focus:not(:disabled) {
  border: 1px solid #1F95EB;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(60%);
  filter: grayscale(100%) opacity(60%);
}

<select
  class="c0"
/>
`;

exports[`Select Component should match snapshot with darkTheme 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #303030;
  background: #505050;
  padding: 0.5rem 1.6rem 0.5rem 0.5rem;
  border-radius: 0;
  background-image: url("data:image/svg+xml,%3Csvg width='13' height='13' viewBox='0 0 10 10' fill='none' xmlns='http://www.w3.org/2000/svg'%3E %3Cpath d='M9.48486 4.24268L5.24219 8.48535L0.999512 4.24268' stroke='%23909090'/%3E %3C/svg%3E ");
  background-repeat: no-repeat;
  background-position: right 0.5rem top calc(50% - 1px);
  -webkit-transition: filter 0.05s;
  transition: filter 0.05s;
}

.c0:focus:not(:disabled) {
  border: 1px solid #1F95EB;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(60%);
  filter: grayscale(100%) opacity(60%);
}

<select
  class="c0"
/>
`;

exports[`Select Component should match snapshot with lightTheme 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #909090;
  background: #FFFFFF;
  padding: 0.5rem 1.6rem 0.5rem 0.5rem;
  border-radius: 0;
  background-image: url("data:image/svg+xml,%3Csvg width='13' height='13' viewBox='0 0 10 10' fill='none' xmlns='http://www.w3.org/2000/svg'%3E %3Cpath d='M9.48486 4.24268L5.24219 8.48535L0.999512 4.24268' stroke='%23909090'/%3E %3C/svg%3E ");
  background-repeat: no-repeat;
  background-position: right 0.5rem top calc(50% - 1px);
  -webkit-transition: filter 0.05s;
  transition: filter 0.05s;
}

.c0:focus:not(:disabled) {
  border: 1px solid #1F95EB;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(60%);
  filter: grayscale(100%) opacity(60%);
}

<select
  class="c0"
/>
`;
